src/test/java/net/imagej/legacy/plugin/IJ1MacroLanguageNamedObjectTest.java

```diff
@@ -0,0 +1,133 @@
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package net.imagej.legacy.plugin;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.scijava.Context;
+import org.scijava.object.ObjectService;
+import org.scijava.script.ScriptLanguage;
+import org.scijava.script.ScriptModule;
+import org.scijava.script.ScriptService;
+
+import javax.script.Bindings;
+import javax.script.ScriptContext;
+import javax.script.ScriptEngine;
+import javax.script.ScriptException;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.Assert.*;
+
+/**
+ * Tests whether {@link IJ1MacroLanguage} works conveniently with named Objects present in the {@link ObjectService}
+ *
+ * This test works in combination with {@link MacroRecorderPostprocessorNamedObjectTest}
+ *
+ * Extra : groovy : should return the Object while in IJ1
+ *
+ * TODO : how to deal with duplicate names ? In terms of testing ?
+ *
+ * @author Nicolas Chiaruttini
+ */
+
+public class IJ1MacroLanguageNamedObjectTest {
+
+	private Context context;
+	private ScriptService scriptService;
+	private ObjectService objectService;
+
+	@Before
+	public void setUp() {
+		context = new Context();
+		scriptService = context.service(ScriptService.class);
+		objectService = context.service(ObjectService.class);
+	}
+
+	@After
+	public void tearDown() {
+		context.dispose();
+	}
+
+	@Test
+	public void testIJ1MacroLanguageNamedObject() throws InterruptedException,
+			ExecutionException
+	{
+		// Puts a named Pet object into the ObjectService
+		Pet pet = new Pet("Felix", "Cat");
+		objectService.addObject(pet,pet.getName());
+
+		// Makes Pet class easily discoverable be the script service
+		scriptService.addAlias(Pet.class);
+
+		// Execute a script.
+		final String script = "" + //
+				"#@ Pet animal\n" + //
+				"#@output String greeting\n" + //
+				"greeting = 'Oh, '+animal+' is so cute!'\n" + //
```

Comment:
What is the benefit of having the object name available in the macro language? Apart from printing log messages, I don't see much value in this, as there's no way to actually *do* anything with the input object. IMHO it would only make sense if you use the API of the object, i.e. for other scripting languages than IJ1 macros.

Reply:
Imagine you have an IJ1 script that manipulate a BdvHandle, and a library with a String to BdvHandle converter. The  user makes a screenshot and close the window.
The recorder then outputs:
```
run("BdvScreenShot", "bdvh=BDV_01");
run("BdvClose", "bdvh=BDV_01");
```
Now the user wants to make a small script that does both actions. And he likes the macro language.
He will just have to do a script like that:
```
#@BdvHandle bdvh;
run("BdvScreenShot", "bdvh="+bdvh);
run("BdvClose", "bdvh="+bdvh);
```

Does that make sense ? I wanted to try it but I cannot use the new legacy I built because of the error I mentioned on gitter, but I have no time to investigate it now. Will later later or next week.

Reply:
I tried and it works.